src/main/java/frc/robot/hardware/rev/request/SparkMaxRequest.java

```diff
@@ -33,6 +35,10 @@ public class SparkMaxRequest<T> implements IRequest<T> {
 
 	@Override
 	public SparkMaxRequest<T> withSetPoint(T setPoint) {
+		if (controlType == CANSparkBase.ControlType.kVelocity) {
+			this.setPoint = (T) Rotation2d.fromRotations(Conversions.perSecondToPerMinute(((Rotation2d) setPoint).getRotations()));
```

Comment:
does velocity always refer to rotation2d?

Reply:
yes

Reply:
i mean couldn't it also be a double when you want to measure the velocity of an elevator?

Reply:
1. the elevator works with position not velocity
2. the elevator converts meters(double) to rotations(Rotation2d), take a look at Raful's elevator

Reply:
this class is being used for motor, motors can only spin and any "toMeters" logic should be out of the hardware handle